internal/handlers/common/filter.go

```diff
@@ -56,7 +56,11 @@ func FilterDocument(doc, filter *types.Document) (bool, error) {
 func filterDocumentPair(doc *types.Document, filterKey string, filterValue any) (bool, error) {
 	if strings.ContainsRune(filterKey, '.') {
 		// {field1./.../.fieldN: filterValue}
-		path := types.NewPathFromString(filterKey)
+		path, err := types.NewPathFromString(filterKey)
+		if err != nil {
+			return false, err
```

Comment:
I belive with the current implementation for `NewPathFromString` we should return lazyerrors everywhere (like in `distinct`) to trace more details. 

Or otherwise we should add tests with wrong paths and see what "compat" returns and return the same.

Reply:
Added call to `lazyerrors.Error` when processing `NewPathFromString` errors.